### PR TITLE
[JENKINS-41729] Convert core `.properties` files to UTF-8

### PR DESCRIPTION
**Pull request by basil**
See [JENKINS-41729](https://issues.jenkins.io/browse/JENKINS-41729). This PR applies an [automated conversion script](https://gist.github.com/basil/c4dc312c19c6ddc80d03275a5ed0f484) to convert all core properties files to UTF-8.

We currently use `maven-encoding-plugin` to enforce encoding. This seems to be another abandoned Kohsuke project; I can't even find it on GitHub. Rather than try to resurrect this dead project I simply removed `maven-encoding-plugin` from our build. Future work could be to add enforcement of UTF-8, but I think this PR is good enough for now.

Recent versions of the Jenkins test harness have been throwing an `AssertionError` if plugins contain a `.properties` file which is both valid UTF-8 and valid ISO-8859-1; this check was relaxed for plugins with a baseline of 2.357 or later in https://github.com/jenkinsci/jenkins-test-harness/pull/448.

### Testing done

I tested the UI with the Locale plugin in `pt_BR`, `ja`, and `de`; everything looked fine.

### Proposed changelog entries

Developer: Jenkins core `.properties` files are now encoded in UTF-8.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since="TODO")` or `@Deprecated(since="TODO", forRemoval=true)` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6719"></a>

### CODE REVIEW

File: test/pom.xml [hunk 1]
```diff
@@ -90,7 +90,8 @@ THE SOFTWARE.
     <dependency>
       <groupId>${project.groupId}</groupId>
       <artifactId>jenkins-test-harness</artifactId>
-      <version>1784.v83a_b_3ce99392</version>
+      <!-- TODO https://github.com/jenkinsci/jenkins-test-harness/pull/448 -->
```
Comment (basil) [inline test/pom.xml:93]: https://github.com/jenkinsci/jenkins-test-harness/pull/448

**Comment by timja** [comment]
I looked through a few diffs and then just checked the non properties file changes, looks great

```
➜  jenkins git:(unicode-properties) git diff --stat master | grep -v properties
 core/pom.xml                                       |   13 -
 test/pom.xml                                       |    2 +-
 ...ellyTest.java => CoreAutomaticTestBuilder.java} |   18 +-
 test/src/test/java/jenkins/CorePropertiesTest.java |   42 -
 5343 files changed, 19983 insertions(+), 20026 deletions(-)
```

One issue though is that IDEs don't handle this very well out of the box, e.g. see IntelliJ:

image

This could be fixed for IntelliJ by either including docs with a reference to https://www.jetbrains.com/help/idea/properties-files.html#encoding

or including this config:

```diff
diff --git a/.gitignore b/.gitignore
index 2c07259eb8..021201b717 100644
--- a/.gitignore
+++ b/.gitignore
@@ -8,6 +8,7 @@ work
 /.idea/*
 !/.idea/icon.svg
 !/.idea/.name
+!/.idea/encodings.xml
 out
 
 # Eclipse and VSCode project files
diff --git a/.idea/encodings.xml b/.idea/encodings.xml
new file mode 100644
index 0000000000..51608ee687
--- /dev/null
+++ b/.idea/encodings.xml
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="Encoding" defaultCharsetForPropertiesFiles="UTF-8">
+    <file url="file://$PROJECT_DIR$/cli/src/filter/resources" charset="UTF-8" />
+    <file url="file://$PROJECT_DIR$/cli/src/main/java" charset="UTF-8" />
+    <file url="file://$PROJECT_DIR$/cli/src/main/resources" charset="UTF-8" />
+    <file url="file://$PROJECT_DIR$/core/src/filter/resources" charset="UTF-8" />
+    <file url="file://$PROJECT_DIR$/core/src/main/java" charset="UTF-8" />
+    <file url="file://$PROJECT_DIR$/core/src/main/resources" charset="UTF-8" />
+    <file url="file://$PROJECT_DIR$/war/src/main/resources" charset="UTF-8" />
+  </component>
+</project>
\ No newline at end of file
```

For NetBeans the only doc I've seen is https://bz.apache.org/netbeans/show_bug.cgi?id=75906 although @jglick may know better

**Comment by jglick** [comment]
I am not aware offhand of any way to convince NetBeans that the files are in UTF-8. Probably best left to non-Java IDEs to edit these files now, ironically.

**Comment by basil** [comment]
> or including this config

I have included this configuration.

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Comment by h1dden-da3m0n** [comment]
random, but would it not be worth it to include the [charset](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#charset) property in the already present `.editorconfig`.
This would lead to every file created after this PR to be in `utf-8`, it can be scoped to property files via a file extension filter and is supported by default by a multitude of IDEs.

Just thought this might be relevant/useful here :thinking:

**Pull request merged**